otherlibs/systhreads/st_stubs.c

```diff
@@ -403,8 +403,8 @@ static void caml_thread_remove_info(caml_thread_t th)
 #ifndef NATIVE_CODE
   stat_free(th->stack_low);
 #endif
-  if (th->backtrace_buffer != NULL) free(th->backtrace_buffer);
 #ifndef WITH_SPACETIME
+  if (th->backtrace_buffer != NULL) caml_stat_free(th->backtrace_buffer);
```

Comment:
Wow. No, it's a silly blunder on my side that happened during a rebase. Thanks for spotting. Fixed it.